_pages/dev/performance/core-web-vitals/cumulative-layout-shift.md

```diff
@@ -0,0 +1,257 @@
+
+When you're using Spartacus Server-Side Rendering (which is recommended for various reasons, including a good Core Web Vital metric Largest Content Paint), then to avoid layout shifts, we should not change the DOM structure with JavaScript after the page has loaded. In particular it's not recommended to use the Spartacus Javascript-based `BreakpointService` to dynamically change the layout of the page. It's because first the Server-Side rendered page is displayed based on the rendered HTML and CSS, but the Javascript is loaded only after a delay and when it changes the layout, it can negatively impact the Cumulative Layout Shift metric.
+
+Ideally the responsive layout should be controlled just with the static HTML and CSS, but not changed with lately-loaded Javascript.
+
+### Why CSS rules are preferred over JavaScript for responsive layout
+[Angular SSR native feature of inlining critical CSS](https://angular.dev/reference/configs/workspace-config#styles-optimization-options) is responsible for ensuring that the CSS is inlined in the `<head>` of the Server-Side rendered HTML and therefore it's applied by the browser immediately when loading the SSR HTML. Thanks to this, the visual layout is correct from the beginning user seeing the page. This is done automatically by Angular SSR and does not require any additional configuration in Spartacus.
+
+### Recent Spartacus improvements to avoid layout shifts
+
+#### Avoid header layout shift
+Before v2211.43, Spartacus default layout of the header and top navigation on desktop viewport was shifted after the lately-loaded Javascript caused the layout to change. This issue is fixed since Spartacus v2211.43, when the feature toggle `unifiedDefaultHeaderSlotsAcrossBreakpoints` is enabled.
+
+**Important**: Additionally, for apps created before v2211.43 you'll need to also change the deprecated `provideConfig(layoutConfig)` to `provideConfigFactory(layoutConfigFactory)` in your `spartacus-features.module.ts`. 
+
+For more on this issue, read the section below [Appendix: Use the same Header layout configuration for all breakpoints](#Use-the-same-Header-layout-configuration-for-all-breakpoints).
+
+### Use the improved product carousel implementation available since v2211.43
+Before v2211.43, Spartacus used the `<cx-carousel>` component to display product carousels. This carousel implementation caused layout shifts when transitioning from Server-Side Rendered (SSR) HTML to Client-Side Rendered (CSR) HTML, as it changed the DOM structure after the JavaScript was loaded and executed on viewport breakpoints other than mobile. This issue is fixed since Spartacus v2211.43, when the feature toggle `productCarouselScrolling` is enabled.
+
+When `productCarouselScrolling` is enabled, the improved carousel implementation `<cx-carousel-scrolling>` is used instead of the old `<cx-carousel>` as a child of the in the `ProductCarouselComponent` and `ProductReferencesComponent`.
```

Comment:
```suggestion
When `productCarouselScrolling` is enabled, the improved carousel implementation `<cx-carousel-scrolling>` is used instead of the old `<cx-carousel>` as a child of the `ProductCarouselComponent` and `ProductReferencesComponent`.
```